cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx

```diff
@@ -0,0 +1,181 @@
+        backgroundColor: theme.palette.grey[700],
+        '& $requestActionButton': {
+          visibility: 'visible',
+        },
+      },
+    },
+    requestMethod: {
+      paddingLeft: '5px',
+      color: theme.palette.white[50],
+      height: '100%',
+      fontWeight: 800,
+      justifyContent: 'center',
+      alignItems: 'flex-start',
+      fontSize: '10px',
+      width: '10%',
+    },
+    requestMethodText: {
+      width: '100%',
+      textAlign: 'left',
+      alignSelf: 'center',
+    },
+    requestPath: {
+      maxWidth: '80%',
+      minWidth: '80%',
+      wordWrap: 'break-word',
+      textAlign: 'left',
+      alignSelf: 'center',
+      textTransform: 'lowercase',
+      fontSize: '10px',
+      display: 'inline-block',
+      lineHeight: '1.3',
+    },
+    getMethod: {
+      color: theme.palette.green[400],
+    },
+    postMethod: {
+      color: theme.palette.yellow[50],
+    },
+    putMethod: {
+      color: theme.palette.blue[50],
+    },
+    deleteMethod: {
+      color: theme.palette.red[50],
+    },
+    buttonTooltip: {
+      fontSize: '14px',
+      backgroundColor: theme.palette.grey[50],
+    },
+    requestActionButtons: {
+      justifyContent: 'flex-end',
+      whiteSpace: 'nowrap',
+    },
+    requestActionButton: {
+      display: 'inline-block',
+      visibility: 'hidden',
+    },
+  };
+};
+
+const mapDispatch = (dispatch) => {
+  return {
+    onRequestClick: (request: IRequestHistory) => {
+      dispatch({
+        type: HttpExecutorActions.setRequestHistoryView,
+        payload: request,
+      });
+    },
+  };
+};
+
+interface IRequestRowProps extends WithStyles<typeof styles> {
+  request: IRequestHistory;
+  onRequestClick: (request: IRequestHistory) => void;
+}
+
+const RequestRowView: React.FC<IRequestRowProps> = ({ classes, request, onRequestClick }) => {
+  const [deleteDialogOpen, setDeleteDialogOpen] = React.useState(false);
+
+  const renderTimestamp = (timestamp: Date) => {
+    const dateOptions = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+    const timeOptions = {
+      hour12: true,
+      hour: '2-digit',
+      minute: '2-digit',
+      second: '2-digit',
+    };
+
+    return `${timestamp.toLocaleDateString('en-US', dateOptions)} ${timestamp.toLocaleTimeString(
+      'en-US',
+      timeOptions
+    )}`;
+  };
+
+  return (
+    <div>
+      <div className={classes.requestRow} onClick={() => onRequestClick(request)}>
+        <div className={classes.requestTimestamp}>
+          <Tooltip
+            classes={{
+              tooltip: classes.buttonTooltip,
+            }}
+            title={renderTimestamp(request.requestID)}
+            placement="right"
+            className={classes.requestActionButton}
+          >
+            <ScheduleIcon />
+          </Tooltip>
+        </div>
+        <div
+          className={classnames(classes.requestMethod, {
+            [classes.getMethod]: request.method === RequestMethod.GET,
```

Comment:
use enum?

Reply:
Which variable are you referring to?

Reply:
I meant the request methods, but it looks like you are already doing it so maybe I was looking in the wrong place or saw something outdated. So no change needed!